persistent-postgresql/Database/Persist/Postgresql.hs

```diff
@@ -1738,6 +1761,310 @@ repsertManySql ent n = putManySql' conflictColumns fields ent n
     fields = keyAndEntityFields ent
     conflictColumns = escapeF . fieldDB <$> entityKeyFields ent
 
+-- | This type is used to determine how to update rows using Postgres'
+-- @INSERT ... ON CONFLICT KEY UPDATE@ functionality, exposed via
+-- 'upsertWhere' and 'upsertManyWhere' in this library.
+--
+-- @since 2.12.0.1
+data HandleUpdateCollision record where
```

Comment:
copied from `MySQL.hs`, should I just import it instead?

Reply:
We don't want to import it, since this package is only for postgres stuff, and we don't want to depend on `persistent-mysql`. But we may want to upstream the definition to `persistent` somewhere so it can be shared. I'm happy to leave that as a later refactoring task - it's entirely possible that the two datatypes will need to diverge to support platform-specific features.